src/app/test/service-stubs.ts

```diff
@@ -83,6 +83,9 @@ export class SourceFileTabsStubService {
   getDescriptorPath(descriptorType: ToolDescriptor.TypeEnum, filePath: string, versionName: string): string {
     return '';
   }
+  isPrimaryDescriptor(path): boolean {
```

Comment:
I don't think you need this anymore, because you removed `isPrimaryDescriptor` from SourceFileTabsService.